internal/pkg/cli/svc_package.go

```diff
@@ -211,13 +203,14 @@ func (o *packageSvcOpts) Execute() error {
 	if err := o.writeAndClose(o.paramsWriter, svcTemplates.configuration); err != nil {
 		return err
 	}
-	addonsTemplate, err := o.getAddonsTemplate()
-	// return nil if addons not found.
-	var notFoundErr *addon.ErrAddonsNotFound
-	if errors.As(err, &notFoundErr) {
-		return nil
-	}
+	addonsTemplate, err := generator.AddonsStackBuilder().Template()
```

Comment:
I assume we are returning a `StackBuilder` interface to make unit-testing easier. An alternative that I've seen before is to provide a fake constructor to make testing easier for other packages (similar to [`afero.MemMapFs`](https://pkg.go.dev/github.com/spf13/afero#NewMemMapFs))

My suggestion is to do something like this:
```go
package addon

// MemAddons creates an *Addons object using dummy template and params values instead of reading the files from the workspace.
func MemAddons(tpl, params string) *Addons {
   return &Addons{
      cachedTemplate: tpl,
      params: params,
   }
}
```

Then in your unit tests:
```go
mockGenerator.EXPECT().Addons().Return(addons.MemAddons("mytemplate", "myparams"), nil)
```


This pattern is also mentioned in the [software engineering at google book](https://abseil.io/resources/swe-book/html/ch13.html#faking-id00042).

Reply:
should be resolved by https://github.com/aws/copilot-cli/pull/3811#discussion_r929391909